src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -145,13 +145,17 @@ private function parseDefinition($id, \DOMElement $service, $file)
             $definition = new Definition();
         }
 
-        foreach (array('class', 'scope', 'public', 'factory-class', 'factory-method', 'factory-service', 'synthetic', 'synchronized', 'lazy', 'abstract') as $key) {
+        foreach (array('class', 'scope', 'public', 'factory-class', 'factory-method', 'factory-service', 'synthetic', 'lazy', 'abstract') as $key) {
             if ($value = $service->getAttribute($key)) {
                 $method = 'set'.str_replace('-', '', $key);
                 $definition->$method(XmlUtils::phpize($value));
             }
         }
 
+        if ($value = $service->getAttribute('synchronized')) {
+            $definition->setSynchronized(XmlUtils::phpize($value), false);
```

Comment:
the second argument should be not be used here. We want other bundles to be notified if they define a synchronized service. FrameworkBundle should handle it it by setting it as synchronized in the DI extension rather than in the XML file to be able to ignore the deprecation warning